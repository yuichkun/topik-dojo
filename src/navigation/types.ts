/**
 * ナビゲーション型定義
 */

import type { NativeStackScreenProps } from '@react-navigation/native-stack';

// ルートスタックのパラメータ定義
export type RootStackParamList = {
  Top: undefined;
  LearningModeSelection: {
    level: number;
  };
  UnitSelection: { 
    level: number; 
  };
  Learning: { 
    level: number; 
    unitNumber: number; 
  };
  TestModeSelection: { 
    level: number; 
  };
  // TODO: 他の画面のパラメータ定義を追加
  // Results: { level: number };
  // ListeningUnitSelection: { level: number; mode: 'listening' };
  // ReadingUnitSelection: { level: number; mode: 'reading' };
  // ListeningTest: { level: number; unitId: string };
  // ReadingTest: { level: number; unitId: string };
  // Review: undefined;
};

// 各画面のプロップス型定義
export type TopScreenProps = NativeStackScreenProps<RootStackParamList, 'Top'>;
export type LearningModeSelectionScreenProps = NativeStackScreenProps<RootStackParamList, 'LearningModeSelection'>;
export type UnitSelectionScreenProps = NativeStackScreenProps<RootStackParamList, 'UnitSelection'>;
export type LearningScreenProps = NativeStackScreenProps<RootStackParamList, 'Learning'>;
export type TestModeSelectionScreenProps = NativeStackScreenProps<RootStackParamList, 'TestModeSelection'>;

declare global {
  namespace ReactNavigation {
    interface RootParamList extends RootStackParamList {}
  }
}