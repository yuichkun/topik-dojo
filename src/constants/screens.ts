/**
 * 画面名の定数定義
 */

export const SCREEN_NAMES = {
  TOP: 'Top',
  LEARNING_MODE_SELECTION: 'LearningModeSelection',
  UNIT_SELECTION: 'UnitSelection',
  LEARNING: 'Learning',
  TEST_MODE_SELECTION: 'TestModeSelection',
  TEST_UNIT_SELECTION: 'TestUnitSelection',
  READING_TEST: 'ReadingTest',
  LISTENING_TEST: 'ListeningTest',
  REVIEW: 'Review',
  // TODO: 他の画面名を追加
  // RESULTS: 'Results',
} as const;