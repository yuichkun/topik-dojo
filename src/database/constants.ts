// テーブル名の定数定義（型安全性向上）
export enum TableName {
  UNITS = 'units',
  WORDS = 'words',
  SRS_MANAGEMENT = 'srs_management',
  LEARNING_STATUS = 'learning_status',
  TEST_RESULTS = 'test_results',
  TEST_QUESTIONS = 'test_questions',
  REVIEW_HISTORY = 'review_history',
  LEARNING_PROGRESS = 'learning_progress',
}

// データベース設定の定数定義
export const DATABASE_CONFIG = {
  name: 'TopikDojo',
  testName: ':memory:',
} as const;