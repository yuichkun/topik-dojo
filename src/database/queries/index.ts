// ユニット関連のクエリ
export * from './unitQueries';

// 単語関連のクエリ
export * from './wordQueries';