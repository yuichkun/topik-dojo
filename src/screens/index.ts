/**
 * 画面コンポーネントのエクスポート
 */

export { default as TopScreen } from './TopScreen';
export { default as LearningModeSelectionScreen } from './LearningModeSelectionScreen';
export { default as UnitSelectionScreen } from './UnitSelectionScreen';
export { default as LearningScreen } from './LearningScreen';
export { default as TestModeSelectionScreen } from './TestModeSelectionScreen';
export { default as TestUnitSelectionScreen } from './TestUnitSelectionScreen';
export { default as ReadingTestScreen } from './ReadingTestScreen';
export { default as ListeningTestScreen } from './ListeningTestScreen';
export { default as ReviewScreen } from './ReviewScreen';