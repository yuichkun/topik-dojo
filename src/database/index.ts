import { Database } from '@nozbe/watermelondb';
import SQLiteAdapter from '@nozbe/watermelondb/adapters/sqlite';

import schema from './schema';
import migrations from './migrations';
import { modelClasses } from './models';
import { DATABASE_CONFIG } from './constants';

// SQLiteアダプターの設定
const adapter = new SQLiteAdapter({
  schema,
  migrations,
  jsi: true, // JSI (JavaScript Interface) を有効化（パフォーマンス向上）
  dbName: DATABASE_CONFIG.name, // データベースファイル名を設定から取得（.dbは自動付与される）
});

// データベースインスタンスの作成
const database = new Database({
  adapter,
  modelClasses,
});

export default database;