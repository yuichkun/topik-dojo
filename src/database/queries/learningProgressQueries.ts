import database from '../index';
import { LearningProgress, WordMastery, Word } from '../models';
import { TableName } from '../constants';
import { Q } from '@nozbe/watermelondb';
import { format } from 'date-fns';

/**
 * 指定日の学習進捗データを取得
 * @param date 日付 (YYYY-MM-DD)
 * @param grade 級
 * @returns 学習進捗データ（存在しない場合はnull）
 */
export async function getLearningProgressByDate(
  date: string,
  grade: number
): Promise<LearningProgress | null> {
  try {
    const progress = await database.collections
      .get<LearningProgress>(TableName.LEARNING_PROGRESS)
      .query(
        Q.and(
          Q.where('date', date),
          Q.where('grade', grade)
        )
      )
      .fetch();

    return progress.length > 0 ? progress[0] : null;
  } catch (error) {
    console.error('Failed to get learning progress by date:', error);
    return null;
  }
}

/**
 * 指定級の最近N日間の学習進捗データを取得
 * @param grade 級
 * @param days 取得日数（デフォルト30日）
 * @returns 学習進捗データ配列（日付降順）
 */
export async function getRecentLearningProgress(
  grade: number,
  days: number = 30
): Promise<LearningProgress[]> {
  try {
    const endDate = Date.now();
    const startDate = endDate - (days - 1) * 24 * 60 * 60 * 1000;

    const startDateStr = format(startDate, 'yyyy-MM-dd');
    const endDateStr = format(endDate, 'yyyy-MM-dd');

    const progress = await database.collections
      .get<LearningProgress>(TableName.LEARNING_PROGRESS)
      .query(
        Q.and(
          Q.where('grade', grade),
          Q.where('date', Q.gte(startDateStr)),
          Q.where('date', Q.lte(endDateStr))
        ),
        Q.sortBy('date', Q.desc)
      )
      .fetch();

    return progress;
  } catch (error) {
    console.error('Failed to get recent learning progress:', error);
    return [];
  }
}

/**
 * 現在の習得状況から学習進捗データを作成または更新
 * @param grade 級
 * @param date 日付 (YYYY-MM-DD)、未指定の場合は今日
 * @returns 作成または更新された学習進捗データ
 */
export async function updateOrCreateLearningProgress(
  grade: number,
  date?: string
): Promise<LearningProgress | null> {
  try {
    const targetDate = date || format(Date.now(), 'yyyy-MM-dd');

    // 現在の習得状況を取得
    const listeningMasteredCount = await getListeningMasteredCountForGrade(grade);
    const readingMasteredCount = await getReadingMasteredCountForGrade(grade);
    const totalWordsCount = await getTotalWordsCountForGrade(grade);

    // 既存のレコードを確認
    const existingProgress = await getLearningProgressByDate(targetDate, grade);

    let progress: LearningProgress;

    if (existingProgress) {
      // 既存レコードを更新
      progress = await database.write(async () => {
        return await existingProgress.update((p) => {
          p.listeningMasteredCount = listeningMasteredCount;
          p.readingMasteredCount = readingMasteredCount;
          p.totalWordsCount = totalWordsCount;
        });
      });
    } else {
      // 新規レコードを作成
      progress = await database.write(async () => {
        return await database.collections
          .get<LearningProgress>(TableName.LEARNING_PROGRESS)
          .create((p) => {
            p.date = targetDate;
            p.grade = grade;
            p.listeningMasteredCount = listeningMasteredCount;
            p.readingMasteredCount = readingMasteredCount;
            p.totalWordsCount = totalWordsCount;
          });
      });
    }

    return progress;
  } catch (error) {
    console.error('Failed to update or create learning progress:', error);
    return null;
  }
}

/**
 * 指定級のリスニング習得語彙数を取得
 * @param grade 級
 * @returns 習得済み語彙数
 */
async function getListeningMasteredCountForGrade(grade: number): Promise<number> {
  try {
    const masteredWords = await database.collections
      .get<WordMastery>(TableName.WORD_MASTERY)
      .query(Q.where('test_type', 'listening'))
      .fetch();

    let count = 0;
    for (const mastery of masteredWords) {
      const word = await database.collections
        .get<Word>(TableName.WORDS)
        .find(mastery.wordId);
      
      if (word && word.grade === grade) {
        count++;
      }
    }

    return count;
  } catch (error) {
    console.error('Failed to get listening mastered count:', error);
    return 0;
  }
}

/**
 * 指定級のリーディング習得語彙数を取得
 * @param grade 級
 * @returns 習得済み語彙数
 */
async function getReadingMasteredCountForGrade(grade: number): Promise<number> {
  try {
    const masteredWords = await database.collections
      .get<WordMastery>(TableName.WORD_MASTERY)
      .query(Q.where('test_type', 'reading'))
      .fetch();

    let count = 0;
    for (const mastery of masteredWords) {
      const word = await database.collections
        .get<Word>(TableName.WORDS)
        .find(mastery.wordId);
      
      if (word && word.grade === grade) {
        count++;
      }
    }

    return count;
  } catch (error) {
    console.error('Failed to get reading mastered count:', error);
    return 0;
  }
}

/**
 * 指定級の総語彙数を取得
 * @param grade 級
 * @returns 総語彙数
 */
async function getTotalWordsCountForGrade(grade: number): Promise<number> {
  try {
    const count = await database.collections
      .get<Word>(TableName.WORDS)
      .query(Q.where('grade', grade))
      .fetchCount();

    return count;
  } catch (error) {
    console.error('Failed to get total words count:', error);
    return 0;
  }
}