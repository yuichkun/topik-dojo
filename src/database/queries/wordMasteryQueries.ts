import database from '../index';
import { WordMastery, Word } from '../models';
import { TableName } from '../constants';
import { Q } from '@nozbe/watermelondb';

/**
 * 単語の習得状態を記録する
 * @param wordId 単語ID
 * @param testType テストタイプ ('listening' | 'reading')
 * @returns 作成されたWordMasteryオブジェクト（既に存在する場合はnull）
 */
export async function createWordMastery(
  wordId: string,
  testType: 'listening' | 'reading'
): Promise<WordMastery | null> {
  try {
    // 既に存在するかチェック
    const existing = await database.collections
      .get<WordMastery>(TableName.WORD_MASTERY)
      .query(
        Q.and(
          Q.where('word_id', wordId),
          Q.where('test_type', testType)
        )
      )
      .fetchCount();

    if (existing > 0) {
      return null; // 既に習得済み
    }

    // 新規作成
    const wordMastery = await database.write(async () => {
      return await database.collections
        .get<WordMastery>(TableName.WORD_MASTERY)
        .create((wm) => {
          wm.wordId = wordId;
          wm.testType = testType;
          wm.masteredDate = Date.now();
        });
    });

    return wordMastery;
  } catch (error) {
    console.error('Failed to create word mastery:', error);
    throw error;
  }
}

/**
 * 単語が習得済みかチェックする
 * @param wordId 単語ID
 * @param testType テストタイプ ('listening' | 'reading')
 * @returns 習得済みならtrue
 */
export async function isWordMastered(
  wordId: string,
  testType: 'listening' | 'reading'
): Promise<boolean> {
  try {
    const count = await database.collections
      .get<WordMastery>(TableName.WORD_MASTERY)
      .query(
        Q.and(
          Q.where('word_id', wordId),
          Q.where('test_type', testType)
        )
      )
      .fetchCount();

    return count > 0;
  } catch (error) {
    console.error('Failed to check word mastery:', error);
    return false;
  }
}

/**
 * 特定の級のリスニング習得数を取得
 * @param grade 級
 * @returns 習得済み単語数
 */
export async function getListeningMasteredCount(grade: number): Promise<number> {
  try {
    const masteredWords = await database.collections
      .get<WordMastery>(TableName.WORD_MASTERY)
      .query(Q.where('test_type', 'listening'))
      .fetch();

    // 各word_idに対応する単語の級をチェック
    let count = 0;
    for (const mastery of masteredWords) {
      const word = await database.collections
        .get<Word>(TableName.WORDS)
        .find(mastery.wordId);
      
      if (word && word.grade === grade) {
        count++;
      }
    }

    return count;
  } catch (error) {
    console.error('Failed to get listening mastered count:', error);
    return 0;
  }
}

/**
 * 特定の級のリーディング習得数を取得
 * @param grade 級
 * @returns 習得済み単語数
 */
export async function getReadingMasteredCount(grade: number): Promise<number> {
  try {
    const masteredWords = await database.collections
      .get<WordMastery>(TableName.WORD_MASTERY)
      .query(Q.where('test_type', 'reading'))
      .fetch();

    // 各word_idに対応する単語の級をチェック
    let count = 0;
    for (const mastery of masteredWords) {
      const word = await database.collections
        .get<Word>(TableName.WORDS)
        .find(mastery.wordId);
      
      if (word && word.grade === grade) {
        count++;
      }
    }

    return count;
  } catch (error) {
    console.error('Failed to get reading mastered count:', error);
    return 0;
  }
}