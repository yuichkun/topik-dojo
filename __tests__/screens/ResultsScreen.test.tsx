/**
 * @format
 */

import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import { TouchableOpacity, Text, View } from 'react-native';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import ResultsScreen from '../../src/screens/ResultsScreen';
import { RootStackParamList } from '../../src/navigation/types';
import { SCREEN_NAMES } from '../../src/constants/screens';
import { createTestWords } from '../helpers/databaseHelpers';
import database from '../../src/database';
import { createWordMastery } from '../../src/database/queries/wordMasteryQueries';
import { updateOrCreateLearningProgress } from '../../src/database/queries/learningProgressQueries';
import { format } from 'date-fns';

// Chart componentをモックする
jest.mock('react-native-chart-kit', () => ({
  PieChart: () => 'PieChart',
  LineChart: () => 'LineChart',
}));

const Stack = createNativeStackNavigator<RootStackParamList>();

// Mock screens for navigation
const MockLearningModeSelectionScreen = () => (
  <View>
    <Text>Mock Learning Mode Selection</Text>
  </View>
);
const MockTopScreen = () => (
  <View>
    <Text>Mock Top Screen</Text>
  </View>
);

// テスト用のナビゲーションコンテナーを作成
const createTestNavigationContainer = (
  level: number,
  initialRoute = SCREEN_NAMES.RESULTS,
) => {
  const TestNavigationContainer = () => (
    <NavigationContainer>
      <Stack.Navigator initialRouteName={initialRoute}>
        <Stack.Screen
          name={SCREEN_NAMES.LEARNING_MODE_SELECTION}
          component={MockLearningModeSelectionScreen}
          options={{ headerShown: false }}
        />
        <Stack.Screen
          name={SCREEN_NAMES.RESULTS}
          component={ResultsScreen}
          initialParams={{ level }}
          options={{ headerShown: false }}
        />
        <Stack.Screen
          name={SCREEN_NAMES.TOP}
          component={MockTopScreen}
          options={{ headerShown: false }}
        />
      </Stack.Navigator>
    </NavigationContainer>
  );
  return TestNavigationContainer;
};

describe('ResultsScreen', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders correctly with no data', async () => {
    const TestContainer = createTestNavigationContainer(1);
    let component: ReactTestRenderer.ReactTestRenderer;

    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<TestContainer />);
    });

    expect(component!).toBeDefined();
  });

  test('displays correct header with level', async () => {
    const TestContainer = createTestNavigationContainer(3);
    let component: ReactTestRenderer.ReactTestRenderer;

    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<TestContainer />);
    });

    // ヘッダーのテキストを確認
    const textElements = component!.root.findAllByType(Text);
    const headerText = textElements.find(element => {
      const children = element.props.children;
      return (
        Array.isArray(children) &&
        children[0] === 3 &&
        children[1] === '級 成績確認'
      );
    });
    expect(headerText).toBeDefined();
  });

  test('displays "まだテストを実施していません" when no test data exists', async () => {
    const TestContainer = createTestNavigationContainer(1);
    let component: ReactTestRenderer.ReactTestRenderer;

    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<TestContainer />);
    });

    const noDataText = component!.root.findByProps({
      children: 'まだテストを実施していません',
    });
    expect(noDataText).toBeDefined();
  });

  test('displays mastery data when test results exist', async () => {
    // テスト用のデータを作成
    const words = await createTestWords(database, [
      { korean: '안녕하세요', japanese: 'こんにちは', grade: 2 },
      { korean: '감사합니다', japanese: 'ありがとうございます', grade: 2 },
      { korean: '죄송합니다', japanese: 'すみません', grade: 2 },
      { korean: '괜찮아요', japanese: '大丈夫です', grade: 2 },
    ]);

    // WORD_MASTERYレコードを作成（2語をリスニング、1語をリーディングで習得済み）
    await createWordMastery(words[0].id, 'listening');
    await createWordMastery(words[1].id, 'listening');
    await createWordMastery(words[2].id, 'reading');

    // 学習進捗データを作成
    await updateOrCreateLearningProgress(2);

    const TestContainer = createTestNavigationContainer(2);
    let component: ReactTestRenderer.ReactTestRenderer;

    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<TestContainer />);
    });

    // 習得率のセクションタイトルが表示されることを確認
    const masteryTitle = component!.root.findByProps({ children: '習得率' });
    expect(masteryTitle).toBeDefined();

    // リスニングとリーディングのラベルが表示されることを確認
    const textElements = component!.root.findAllByType(Text);

    // リスニングのラベルを確認 (2/4 = 50%)
    const listeningLabel = textElements.find(element => {
      const children = element.props.children;
      return (
        Array.isArray(children) &&
        children[0] === 'リスニング (' &&
        children[1] === 2 &&
        children[2] === '/' &&
        children[3] === 4 &&
        children[4] === ')'
      );
    });
    expect(listeningLabel).toBeDefined();

    // リーディングのラベルを確認 (1/4 = 25%)
    const readingLabel = textElements.find(element => {
      const children = element.props.children;
      return (
        Array.isArray(children) &&
        children[0] === 'リーディング (' &&
        children[1] === 1 &&
        children[2] === '/' &&
        children[3] === 4 &&
        children[4] === ')'
      );
    });
    expect(readingLabel).toBeDefined();
  });

  test('displays progress chart when daily data exists', async () => {
    // テスト用のデータを作成
    const words = await createTestWords(database, [
      { korean: '안녕하세요', japanese: 'こんにちは', grade: 1 },
      { korean: '감사합니다', japanese: 'ありがとうございます', grade: 1 },
    ]);

    // WORD_MASTERYレコードを作成
    await createWordMastery(words[0].id, 'listening');

    // 複数日の学習進捗データを作成
    await updateOrCreateLearningProgress(1);
    const yesterday = format(Date.now() - 24 * 60 * 60 * 1000, 'yyyy-MM-dd');
    await updateOrCreateLearningProgress(1, yesterday);

    const TestContainer = createTestNavigationContainer(1);
    let component: ReactTestRenderer.ReactTestRenderer;

    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<TestContainer />);
    });

    // 進捗グラフのタイトルが表示されることを確認
    const progressTitle = component!.root.findByProps({
      children: '習得進捗',
    });
    expect(progressTitle).toBeDefined();
  });

  test('back button is present', async () => {
    const TestContainer = createTestNavigationContainer(1);
    let component: ReactTestRenderer.ReactTestRenderer;

    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<TestContainer />);
    });

    // 戻るボタンを確認
    const backButton = component!.root.findByProps({ children: '← 戻る' });
    expect(backButton).toBeDefined();

    // 戻るボタンが押せることを確認
    const touchableOpacities = component!.root.findAllByType(TouchableOpacity);
    const backTouchable = touchableOpacities.find(button => {
      try {
        const text = button.findByProps({ children: '← 戻る' });
        return text !== undefined;
      } catch {
        return false;
      }
    });

    expect(backTouchable).toBeDefined();
    expect(backTouchable!.props.onPress).toBeDefined();
  });

  test('displays retry button functionality', async () => {
    const TestContainer = createTestNavigationContainer(1);
    let component: ReactTestRenderer.ReactTestRenderer;

    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<TestContainer />);
    });

    // 再試行ボタンがエラー状態でのみ表示されるため、ここでは基本的な構造をテスト
    // コンポーネントが正常にレンダリングされることを確認
    expect(component!.root).toBeDefined();
  });

  test('matches snapshot', async () => {
    const TestContainer = createTestNavigationContainer(1);
    let component: ReactTestRenderer.ReactTestRenderer;

    await ReactTestRenderer.act(async () => {
      component = ReactTestRenderer.create(<TestContainer />);
    });

    expect(component!.toJSON()).toMatchSnapshot();
  });
});
