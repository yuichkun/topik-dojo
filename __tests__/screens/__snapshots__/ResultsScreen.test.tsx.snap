// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsScreen matches snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      activityState={2}
      aria-hidden={false}
      collapsable={false}
      fullScreenSwipeShadowEnabled={true}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      hasLargeHeader={false}
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onHeaderHeightChange={[Function]}
      onNativeDismissCancelled={[Function]}
      onSheetDetentChanged={[Function]}
      onTransitionProgress={[Function]}
      onWillAppear={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents={
        [
          1,
        ]
      }
      sheetCornerRadius={-1}
      sheetElevation={24}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetInitialDetent={0}
      sheetInitialDetentIndex={0}
      sheetLargestUndimmedDetent={-1}
      sheetLargestUndimmedDetentIndex={-1}
      stackPresentation="push"
      style={
        [
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ],
          {
            "zIndex": undefined,
          },
        ]
      }
      swipeDirection="horizontal"
    >
      <RNSScreenContentWrapper
        collapsable={false}
        style={
          [
            {
              "flex": 1,
            },
            [
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ],
          ]
        }
      >
        <RCTSafeAreaView
          className="flex-1 bg-white"
        >
          <View
            className="flex-row items-center justify-between px-4 py-3 bg-white border-b border-gray-200"
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Text
                className="text-blue-500 text-base"
              >
                ← 戻る
              </Text>
            </View>
            <Text
              className="text-lg font-bold text-gray-800"
            >
              1
              級 成績確認
            </Text>
            <View
              className="w-12"
            />
          </View>
          <View
            className="flex-1 justify-center items-center px-4"
          >
            <Text
              className="text-lg text-gray-600 mb-2 text-center"
            >
              まだテストを実施していません
            </Text>
            <Text
              className="text-sm text-gray-500 text-center"
            >
              テストを実施すると成績が表示されます
            </Text>
          </View>
        </RCTSafeAreaView>
      </RNSScreenContentWrapper>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleFontFamily="System"
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleFontFamily="System"
        largeTitleFontWeight="700"
        largeTitleHideShadow={false}
        pointerEvents="box-none"
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "position": "absolute",
            "width": "100%",
          }
        }
        title="Results"
        titleColor="rgb(28, 28, 30)"
        titleFontFamily="System"
        titleFontWeight="600"
        topInsetEnabled={true}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
